main/ZgatewayBT.ino

```diff
@@ -430,12 +437,11 @@ void BLEscan() {
   Log.notice(F("Scan begin" CR));
   BLEDevice::init("");
   BLEScan* pBLEScan = BLEDevice::getScan(); //create new scan
-  MyAdvertisedDeviceCallbacks myCallbacks;
-  pBLEScan->setAdvertisedDeviceCallbacks(&myCallbacks);
+  pBLEScan->setAdvertisedDeviceCallbacks(new MyAdvertisedDeviceCallbacks());
```

Comment:
Is the allocated memory later deallocated?

Reply:
Yes by 
`BLEDevice::deinit(true);`
usage
https://github.com/h2zero/NimBLE-Arduino/blob/d8a535d28ab7af8a4074f719fb87406c4b244a69/src/NimBLEDevice.cpp#L543